core/services/pipeline/keypath_test.go

```diff
@@ -0,0 +1,51 @@
+package pipeline_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+)
+
+func TestKeypath(t *testing.T) {
+	t.Parallel()
+
+	t.Run("can be constructed from a period-delimited string with 2 or fewer parts", func(t *testing.T) {
+		t.Parallel()
+
+		kp, err := pipeline.NewKeypathFromString("")
+		assert.NoError(t, err)
+		assert.Equal(t, pipeline.Keypath{}, kp)
+
+		kp, err = pipeline.NewKeypathFromString("foo")
+		assert.NoError(t, err)
+		assert.Equal(t, pipeline.Keypath{NumParts: 1, Part0: "foo"}, kp)
+
+		kp, err = pipeline.NewKeypathFromString("foo.bar")
+		assert.NoError(t, err)
+		assert.Equal(t, pipeline.Keypath{NumParts: 2, Part0: "foo", Part1: "bar"}, kp)
+	})
+
+	t.Run("wrong keypath", func(t *testing.T) {
+		t.Parallel()
+
+		wrongKeyPath := []string{
+			".",
+			"..",
+			"x.",
+			".y",
+			"x.y.",
+			"x.y.z",
+		}
+
+		for _, keypath := range wrongKeyPath {
+			t.Run("keypath", func(t *testing.T) {
```

Comment:
How about the `keypath` var here?
```suggestion
			t.Run(keypath, func(t *testing.T) {
```

Reply:
Oh this is so smart! Sad I forgot about this.